messaginginapp/src/main/java/io/customer/messaginginapp/ModuleMessagingInApp.kt

```diff
@@ -34,19 +34,31 @@ class ModuleMessagingInApp internal constructor(
 
     private val gistProvider by lazy { diGraph.gistProvider }
 
+    private val logger by lazy { diGraph.logger }
+
     override fun initialize() {
         initializeGist(organizationId)
         setupHooks()
         setupGistCallbacks()
     }
 
     private fun setupGistCallbacks() {
-        gistProvider.subscribeToEvents { deliveryID ->
-            trackRepository.trackInAppMetric(
-                deliveryID = deliveryID,
-                event = MetricEvent.opened
-            )
-        }
+        gistProvider.subscribeToEvents(
+            onMessageShown = { deliveryID ->
+                trackRepository.trackInAppMetric(
```

Comment:
I recommend adding a `logger.debug` statement for onMessageShown, onAction for easier debugging of the SDK. 

```suggestion
               logger.debug("in-app message shown $deliveryID")

                trackRepository.trackInAppMetric(
```

Reply:
Yes, a log statement would more then likely exist when adding a task to the background queue. 

Why I still suggest to add a log statement here is that the point of logs is to help walk you through the code step by step. What if there is a bug in the code between `onMessageShown()` and the background queue code? Having a log statement in `onMessageShown()` would help to help us pinpoint where the bug is. 

Having too many logs can indeed get noisy. From my experience, noisy is much better then not enough logs and not being able to debug the bug as quickly.